### DISCUSSION TITLE
API

### DISCUSSION THREAD

**Discussion by MasterCodeIT** [jglim/CaesarSuite#31] [Q&A]
No issue just a quick question and perhaps a hand. Love the project!

I working on a project via python. Only working with 1 specific ECU and wanted to use a line of code you have in your project as an API to auto-calculate the ecukey. 

Using the DaimlerStandardSecurityAlgo, ECU is MED1775

example:
access_level = "05"
seed = "ABCDEFGH"
ecuKey = ECUSeedKeyDLL(seed)
send ecukey  //--- answer

**Comment by jglim** [comment]
Hello,

Thanks for checking out the project. You may wish to look into ConsoleUnlockECU (Windows: ConsoleUnlockECU.exe), which is UnlockECU as a command line tool. The compiled binaries are available in the [releases](https://github.com/jglim/UnlockECU/releases/tag/v1.1) page.

Set up your definition (e.g. name, parameters) first in `db.json`, then run the command line utility with a format similar to this:
```
ConsoleUnlockECU --database=db.json --name=MED1775 --level=9 --seed=1122334455667788
```
which should then return a hex value like:
```
0453FD7A
```

---

As this isn't an issue, [discussions](https://github.com/jglim/CaesarSuite/discussions) would be a better place to seek community help.

**Comment by MasterCodeIT** [comment]
> 
> 
> Hello,
> 
> Thanks for checking out the project. You may wish to look into ConsoleUnlockECU (Windows: ConsoleUnlockECU.exe), which is UnlockECU as a command line tool. The compiled binaries are available in the [releases](https://github.com/jglim/UnlockECU/releases/tag/v1.1) page.
> 
> Set up your definition (e.g. name, parameters) first in `db.json`, then run the command line utility with a format similar to this:
> 
> ```
> ConsoleUnlockECU --database=db.json --name=MED1775 --level=9 --seed=1122334455667788
> ```
> 
> which should then return a hex value like:
> 
> ```
> 0453FD7A
> ```
> 
> As this isn't an issue, [discussions](https://github.com/jglim/CaesarSuite/discussions) would be a better place to seek community help.

Awesome! So upon doing this, I see MED1775 is already in the DB with a level of 5. 

> 
> 
> Hello,
> 
> Thanks for checking out the project. You may wish to look into ConsoleUnlockECU (Windows: ConsoleUnlockECU.exe), which is UnlockECU as a command line tool. The compiled binaries are available in the [releases](https://github.com/jglim/UnlockECU/releases/tag/v1.1) page.
> 
> Set up your definition (e.g. name, parameters) first in `db.json`, then run the command line utility with a format similar to this:
> 
> ```
> ConsoleUnlockECU --database=db.json --name=MED1775 --level=9 --seed=1122334455667788
> ```
> 
> which should then return a hex value like:
> 
> ```
> 0453FD7A
> ```
> 
> As this isn't an issue, [discussions](https://github.com/jglim/CaesarSuite/discussions) would be a better place to seek community help.

A way to make this cmd work on linux? I see its an exe file.

**Comment by MasterCodeIT** [comment]
A way to make this cmd work on linux? I see its an exe file.

**Reply by jglim** [reply]
Yeah, please check the description on the release again for details: https://github.com/jglim/UnlockECU/releases/tag/v1.1
